### ISSUE TITLE
Allow Front Matter data to override External Data

### ISSUE THREAD

**Issue by fabritw** [zurb/panini#101]
I'm using External Data files to define global variables. In some templates, it would be very helpful to override the globals however (https://github.com/zurb/panini#data).

For example:
```
// src/data/globals.yml - define global variable
url: "www.mysite.com"

// src/pages/signup.html - override global variable
---
globals.url: "signup.mysite.com"
---
```

**Comment by renatodeleao** [comment]
This would be nice . Right now i'm bypassing this with an `if else`, but it feels like extra verbosity.

`{{if url}}{{url}}{{else}}{{globals.url}}{{/if}}`

**Comment by gakimball** [comment]
We could do a deep merge of properties, where now we just overwrite everything if two keys conflict. So you could write:

```yml
globals:
  url: 'signup.mysite.com'
```

That at least prevents us from needing to explain object path notation.

**Comment by gakimball** [comment]
Implemented this in ae6b85352b6adf3dd1d8d1b086f0010cd2fa6823
